Editing the installing CLI (#933)
fishtown-analytics/docs.getdbt.com: website/docs/dbt-cli/install/pip.md

---
title: "Use pip to install dbt"
description: "You can use pip to install dbt Core and adapter plugins from the command line."
---

> You need to use `pip` to instal dbt Core on Windows or Linux operating systems. You should use [Homebrew](../homebrew) for installing dbt Core on a MacOS. 

Author: ```suggestion
You need to use `pip` to instal dbt Core on Windows or Linux operating systems. You should use [Homebrew](install/homebrew) for installing dbt Core on a MacOS. 
```

> You need to use `pip` to instal dbt Core on Windows or Linux operating systems. You should use [Homebrew](install/homebrew) for installing dbt Core on a MacOS. 

Author: ```suggestion
You need to use `pip` to install dbt Core on Windows or Linux operating systems. You should use [Homebrew](install/homebrew) for installing dbt Core on a MacOS. 
```

---
title: "Use pip to install dbt"

> description: ""

Author: ```suggestion
description: "You can use pip to install dbt Core and adapter plugins from the command line."
```

---

> dbt Core and plugins are Python modules distributed on [PyPi](https://pypi.org/project/dbt/), and can be installed via `pip`. We recommend using virtual environments when installing with `pip`.

Author: @jtcohen6 Can you provide a bit about why someone would use pip over Homebrew? Also would you "install from source?" Would it be to get the absolute latest code? And, finally, why would someone use Docker over the others?

Reviewer: Great questions!
- Homebrew only works on Mac, so `pip` is required on Windows, Linux, etc
- Installing from source lets people get the latest code, or to install dbt from a specific commit. In practice, this is relevant for contributing changes, or debugging a past change
- Docker is most useful in production, since it comes with dbt and all of its dependencies already installed (no risk of `pip install dbt-snowflake` failing in the middle of the night because of some borked dependency. To a lesser extent, some folks might use it for local development, if their machine doesn't have a python environment set up, but I'd say that's a lot less common

> dbt Core and plugins are Python modules distributed on [PyPi](https://pypi.org/project/dbt/), and can be installed via `pip`. We recommend using virtual environments when installing with `pip`.

Author: ```suggestion
You need to use `pip` to instal dbt Core on Windows or Linux operating systems. You should use [Homebrew](../homebrew) for installing dbt Core on a MacOS. 

You can install dbt Core and plugins using `pip` because they are Python modules distributed on [PyPi](https://pypi.org/project/dbt/).  We recommend using virtual environments when installing with `pip`.
```

> <FAQ src="install-pip-best-practices" />

Author: @jtcohen6 To decrease the number of FAQs on the FAQ page, what do you think about adding the prerequisites and best practices as sections in this doc instead of FAQs? Or are they referenced in another place? (Also we can punt this until later, just wanted to ask you about it here.)

Reviewer: good conversation out of this about what we should do when there's information relevant (and significant!) for many, but not all, users

  dbt-bigquery
```

> Or, better yet -— install just the one(s) you need!

Author: ```suggestion
Or, better yet, just install the package(s) you need!
```
